### PR TITLE
Initial refactoring to support workload projection

### PR DESCRIPTION
**Pull request by cmoulliard**
- Initial refactoring to support workload projection
- #193

### CODE REVIEW

File: servicebox-app/src/test/java/io/halkyon/ApplicationsPageTest.java [hunk 1]
```diff
@@ -157,16 +157,16 @@ public void testBindApplication(){
...
 
-
         // then secret should have been generated
         String url = serviceName + ":1111";
         String urlBase64 = Base64.getEncoder().encodeToString(url.getBytes());
         String userBase64 = Base64.getEncoder().encodeToString("user1".getBytes());
         String pwdBase64 = Base64.getEncoder().encodeToString("pass1".getBytes());
+        String typeBase64 = Base64.getEncoder().encodeToString("type1".getBytes());
         verify(mockKubernetesClientService, times(1))
                 .mountSecretInApplication(argThat(new ApplicationNameMatcher(appName)),
```
Comment (aureamunoz) [inline servicebox-app/src/test/java/io/halkyon/ApplicationsPageTest.java:167]: You need to add the new param `type`to the [KubernetesClientService#mountSecret method](https://github.com/halkyonio/primaza-poc/blob/d69362960eb59f1f31ef7e91e5db229223657a15/servicebox-app/src/main/java/io/halkyon/services/KubernetesClientService.java#L122)

**Review by aureamunoz — CHANGES_REQUESTED** [review]

**Review by aureamunoz — APPROVED** [review]

**Comment by aureamunoz** [comment]
To be merged @cmoulliard

File: servicebox-app/src/test/java/io/halkyon/ApplicationsPageTest.java (hunk 1 above)
Comment (cmoulliard) [reply servicebox-app/src/test/java/io/halkyon/ApplicationsPageTest.java:167]: Fixed. Many thanks

**Comment by cmoulliard** [comment]
You merged it too early as some changes are still needed to populate correctly the Deployment:
- Set the ENV VAR `SERVICE_BINDING_ROOT = `/binding` 
- Mount the volume of the secret within the container and specify the mountPath
```
        volumeMounts:
        - mountPath: /binding/<SECRET_NAME>
          name: <SECRET_NAME>
```
@aureamunoz

**Pull request merged**